staging/src/k8s.io/component-base/metrics/counter.go

```diff
@@ -45,6 +45,14 @@ func NewCounter(opts *CounterOpts) *Counter {
 	return kc
 }
 
+// Reset resets the underlying prometheus Counter to start counting from 0 again
+func (c *Counter) Reset() {
+	if !c.IsCreated() {
+		return
+	}
+	c.setPrometheusCounter(prometheus.NewCounter(c.CounterOpts.toPromCounterOpts()))
+}
+
```

Comment:
This is not a typical pattern.. there is an implicit contract with process_start_time_seconds that allows the server to maintain the monotonically increasing property of a counter. 

Why do you need the Reset?

Reply:
It doesn't look used anywhere, can we delete this?

Reply:
As discussed, type of TLSHandshakeErrors is changed from CounterVec to Counter, which did not support Reset() method. To keep align with other metircs, I added Reset() method for Counter type. 

This Reset() method is called when requesting DELETE on apiserver /metrics:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/routes/metrics.go#L49